### DISCUSSION TITLE
Can I extend the file extension list in AllowedFile.php?

### DISCUSSION THREAD

**Discussion by jensolafkoch** [statamic/cms#9634] [Q&A] [answered]
I try to upload files to my assets (Statamic 4.51.0) with extensions like ".xml" and ".sdlxliff". They are rejected with a CP message of "The given data is invalid." 

I believe that this is monitored by `vendor/statamic/cms/src/Http/Controllers/CP/Assets/AssetsController.php` and finally the allowed extensions by `vendor/statamic/cms/src/Validation/AllowedFile.php`.

Can I extend this list somehow in a file in `/config/statamic/`?

I need to save whatever filetype it is – preferrably ".*"  ... (It's only for internal use, and I'm the only user.)

**Comment by duncanmcclean** [comment, marked as answer]
Yes, you can add this option to your `assets` config to allow additional file extensions:

https://github.com/statamic/cms/blob/4.x/config/assets.php#L201-L211

**Reply by jensolafkoch** [reply]
Thank you! Sorry, I missed that addition from 4.36 ...
